### PR TITLE
End to end working single pass

### PR DESCRIPTION
**Pull request by cramforce**
- Hooks up the new compilation pass to the existing gulp task with new flag `single_pass`.
- Fixes a first set of issues to due more aggressive compilation.

Known issue: amp-bind isn't working

### CODE REVIEW

File: src/service/extension-location.js [hunk 1]
```diff
@@ -42,8 +42,13 @@ export function calculateExtensionScriptUrl(location, extensionId,
   opt_extensionVersion, opt_isLocalDev) {
   const base = calculateScriptBaseUrl(location, opt_isLocalDev);
   const rtv = getMode().rtvVersion;
-  const extensionVersion = opt_extensionVersion || '0.1';
-  return `${base}/rtv/${rtv}/v0/${extensionId}-${extensionVersion}.js`;
+  if (opt_extensionVersion == null) {
+    opt_extensionVersion = '0.1'
```
Comment (erwinmombay) [inline src/service/extension-location.js]: missing `;`

File: ads/_config.js [hunk 2]
```diff
@@ -64,7 +64,7 @@ let AdNetworkConfigDef;
  * @const {!Object<string, !AdNetworkConfigDef>}}
  */
 export const adConfig = {
-  _ping_: {
+  '_ping_': {
```
Comment (erwinmombay) [inline ads/_config.js]: were these getting obfuscated ?

File: build-system/amp.extern.js [hunk 3]
```diff
@@ -123,6 +123,8 @@ AmpConfigType.prototype.localDev;
 AmpConfigType.prototype.v;
 /* @public {boolean} */
 AmpConfigType.prototype.canary;
+/* @public {string} */
+AmpConfigType.prototype.runtime;
```
Comment (erwinmombay) [inline build-system/amp.extern.js]: where will runtime get used?

Comment (cramforce) [reply build-system/amp.extern.js]: It was just missing. Already being used.

**Comment by cramforce** [comment]
Part of #16847

**Pull request merged**